Dockerfile

```diff
@@ -2,54 +2,111 @@
-    chown -R www-data:www-data /opt/kimai/var && \
-    chown www-data:www-data /opt/kimai/vendor/mpdf/mpdf/tmp
-
-WORKDIR /opt/kimai
+        xsl \
+        xml \
+        iconv \
+        bcmath \
+        curl \
+        fileinfo \
+        gettext \
+        xmlrpc \
+        xmlwriter \
+        simplexml \
+        zip \
+        opcache
+
+# Install Xdebug
+ARG XDEBUG=0
+
+RUN set -ex \
+    && if [ "$XDEBUG" = "1" ]; then \
+      mkdir /tmp/build \
+      && curl -sSL https://github.com/xdebug/xdebug/archive/2.9.8.tar.gz | tar -xzC /tmp/build \
+      && cd /tmp/build/xdebug-* \
+      && phpize \
+      && ./configure --enable-xdebug \
+      && make \
+      && make install \
+      && rm -rf /tmp/build \
+      && echo "zend_extension=xdebug.so" > /usr/local/etc/php/conf.d/xdebug.ini \
+      && echo "xdebug.idekey=PHPSTORM" >> /usr/local/etc/php/conf.d/xdebug.ini \
+      && echo "xdebug.remote_enable=1" >> /usr/local/etc/php/conf.d/xdebug.ini \
+      && echo "xdebug.remote_connect_back=1" >> /usr/local/etc/php/conf.d/xdebug.ini \
+      && echo "xdebug.remote_host=host.docker.internal" >> /usr/local/etc/php/conf.d/xdebug.ini \
+      && echo "xdebug.remote_port=9000" >> /usr/local/etc/php/conf.d/xdebug.ini \
+      && echo "xdebug.remote_autostart=1" >> /usr/local/etc/php/conf.d/xdebug.ini \
+      && echo "xdebug.remote_handler=dbgp" >> /usr/local/etc/php/conf.d/xdebug.ini; \
+    fi
+
+WORKDIR /var/www/html
+
+# Install composer
+ENV COMPOSER_MEMORY_LIMIT=-1
+
+RUN set -ex \
+    && curl -sS https://getcomposer.org/composer-stable.phar -o /usr/local/bin/composer \
+    && chmod +x /usr/local/bin/composer
+
+# Install symfony cli
+RUN set -ex \
+    && curl -sS https://get.symfony.com/cli/installer | bash \
+    && mv /root/.symfony/bin/symfony /usr/local/bin/symfony
+
+# Install production php.ini and make sore there is a folder to hold the demo database
+RUN set -ex \
+    && cp /usr/local/etc/php/php.ini-development /usr/local/etc/php/php.ini \
+    && sed -i 's/memory_limit = 128M/memory_limit = 512M/g' /usr/local/etc/php/php.ini \
+    && sed -i 's/;opcache.memory_consumption=128/opcache.memory_consumption=256/g' /usr/local/etc/php/php.ini \
+    && sed -i 's/;opcache.max_accelerated_files=10000/opcache.max_accelerated_files=20000/g' /usr/local/etc/php/php.ini \
+    && sed -i 's/log_errors_max_len = 1024/log_errors_max_len = 8192/g' /usr/local/etc/php/php.ini \
+    && mkdir -p /var/data \
+    && chown -R www-data:www-data /var/www /var/data
+
+USER www-data:www-data
+
+# Create shallow clone of kimai and configure it as dev role
+RUN set -ex \
+    && git clone --depth 1 https://github.com/kevinpapst/kimai2.git . \
+    && composer install --optimize-autoloader
+
+# Setup demo specific ENV
+ENV APP_ENV=dev \
+    DATABASE_URL=sqlite:////var/data/kimaidemo.sqlite
+
+# Create default SQLite database with a default user and compile it into the
+# image to offer some default setup to spool up incase of a demo.
+RUN set -ex \
```

Comment:
This should be done from outside the container, e.g. in a `docker-compose.yml`.
That would allow to use this container in production, and not come with pre
installed admin account like this.

Reply:
This Dockerfile is, per description, **not** for production and only for development. https://github.com/tobybatch/kimai2 does the publishing towards docker hub and they are perfectly fine for that job. This Dockerfile should be for dev only, using the best and simplest way to allow for fast development/fixing/testing when required.